presto-main/src/main/java/io/prestosql/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1212,7 +1212,7 @@ protected Scope visitTable(Table table, Optional<Scope> scope)
                 if (!metadata.schemaExists(session, new CatalogSchemaName(name.getCatalogName(), name.getSchemaName()))) {
                     throw semanticException(SCHEMA_NOT_FOUND, table, "Schema '%s' does not exist", name.getSchemaName());
                 }
-                throw semanticException(TABLE_NOT_FOUND, table, "Table '%s' does not exist", name);
+                throw semanticException(TABLE_NOT_FOUND, table, "Table '%s' does not exist or is not accessible", name);
```

Comment:
if we want to go in this direction, we would need to update schema check above, and perhaps others

also, this should be test covered (eg to make sure the actual exception comes from here, and not some oither place)